### ISSUE TITLE
Customizable decimal format for AbstractNumberFields

### ISSUE THREAD

**Issue by fof10** [vaadin/flow-components#3925]
### Describe your motivation

`AbstractNumberField` is very useful when a number has to be set within custom borders - specified by `minValue`/`maxValue`.
At the moment, this field uses a predefined, fixed formatter to represent numbers in a specific way.
On the other hand, a `TextField` can display formatted numbers, but offers no functionality for defining `minValue`/`maxValue`.

In cases where big numbers have to be defined within a fixed number range, an adjusted decimal format with digit grouping increases the readability of the number values hence also increases the user experience.

### Describe the solution you'd like

Resulting from the described motivation, it is useful to enhance both `IntegerField `und `NumberField` with a new constructor with a custom decimal format as parameter.
Summing up, an `AbstractNumberField` is capable of both specifying an allowed number range and represent numbers appropriately formatted.

**Comment by TatuLund** [comment]
I made custom DecimalField example using TextFieldFormatter add-on and converters as a workaround.

https://gist.github.com/TatuLund/293ad2cc3a2d10448cbd1630b7464ebf
